lucene/core/src/java/org/apache/lucene/index/AutomatonTermsEnum.java

```diff
@@ -188,7 +188,11 @@ private boolean nextString() {
     savedStates.setIntAt(0, 0);
     
     while (true) {
-      curGen++;
+      if (++curGen == 0) {
+        // Clear the visited states every time curGen overflows (so very infrequently to not impact average perf).
+        curGen++;
```

Comment:
Can we remove this unnecessary increment. Also i'd change the comment from `overflows` to `wraps`.